test/index.test.js

```diff
@@ -302,6 +302,64 @@ describe('GitHub Action Tests', () => {
 
             await run();
 
+            expect(core.setOutput).toHaveBeenCalledWith('execution-id', 'test-execution');
+            expect(core.setOutput).toHaveBeenCalledWith('execution-url', 'http://test.com/execution');
+            expect(global.fetch).toHaveBeenCalledTimes(2); // One for trigger, one for status
+        }, 20000);
+        it('should fail if global "test-config" has an invalid "entrypoint"', async () => {
+            core.getInput = jest.fn().mockImplementation((name) => {
+                if (name === 'suite') return 'test/test-suite';
+                if (name === 'test-config') {
+                    return JSON.stringify({
+                        entrypoint: 123 // Invalid entrypoint
+                    });
+                }
+                return null;
+            });
+
+            await run();
+
+            expect(core.setFailed).toHaveBeenCalledWith(expect.stringContaining('"entrypoint" must be a string if provided. Found number.'));
+        });
+
+        it('should fail if global "test-config" has an invalid "variables" type', async () => {
+            core.getInput = jest.fn().mockImplementation((name) => {
+                if (name === 'suite') return 'test/test-suite';
+                if (name === 'test-config') {
+                    return JSON.stringify({
+                        variables: 'invalid' // Invalid variables type
+                    });
+                }
+                return null;
+            });
+
+            await run();
+
+            expect(core.setFailed).toHaveBeenCalledWith(expect.stringContaining('"variables" must be an object if provided. Found string.'));
+        });
+
+        it('should execute successfully with valid global "test-config"', async () => {
+            core.getInput = jest.fn().mockImplementation((name) => {
+                if (name === 'suite') return 'test/test-suite';
+                if (name === 'stories') {
+                    return JSON.stringify([
+                        {
+                            slug: 'story1',
+                            'test-config': { entrypoint: 'http://example.com/entry1' }
+                        }
+                    ]);
+                }
+                if (name === 'test-config') {
+                    return JSON.stringify({
+                        entrypoint: 'http://example.com/global-entry',
+                        variables: { globalKey: 'globalValue' }
+                    });
+                }
+                return null;
+            });
+
+            await run();
+
             expect(core.setOutput).toHaveBeenCalledWith('execution-id', 'test-execution');
             expect(core.setOutput).toHaveBeenCalledWith('execution-url', 'http://test.com/execution');
             expect(global.fetch).toHaveBeenCalledTimes(2); // One for trigger, one for status
```

Comment:
given the title of this PR, it would make sense to also check that both `test-config` props were correctly taken into account.

e.g. by:
- triggering the run of a story, with a top-level `test-config` that overrides its entrypoint
- => assert that the story was run using the overidded entrypoint, instead of the one specified in the story

Reply:
we do "assert that the story was run using the overidded entrypoint, instead of the one specified in the story" in api.js in backend, there (github action), I just make sure the format is correctly before sending a request to api.js

Reply:
i've updated and added test in test-action.ymal and make sure the the entrypoint and variable is overided. is it okay?

Reply:
The override logic is in api.js. Here, I just make sure the request body format is correct.